### PR TITLE
Bug: fix windows curl

### LINKED ISSUE #123: Support curl dependency on Windows
**Expected Behavior**
Under Windows, the curl dependency should be satisfied and the builds should complete. This should work in both then desktop builds and the CI builds. 

**Current Behavior**
The curl dependency is not handled. 

**Possible Solution**
Either have cmake auto install the dependency, prompt the user before installing, or direct the user as to how to satisfy the dependency.

### PR DESCRIPTION
**Pull request by ciband**
Added additional flags to explicitly set linking with static CRT as well as producing and linking with static libraries.

Fixes #123 

## What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactor
- [ ] Performance
- [ ] Tests
- [x] Build
- [ ] Documentation
- [ ] Code style update
- [ ] Continuous Integration
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Does this PR release a new version?

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch
- [x] All tests are passing
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -47,14 +47,18 @@ include_directories(${PROJECT_SOURCE_DIR})
 set(BUILD_TESTING OFF)
 set(HTTP_ONLY ON)
 set(CMAKE_USE_LIBSSH2 OFF)
+set(CURL_STATICLIB true)
```
Comment (sleepdefic1t) [inline src/CMakeLists.txt]:
```cmake
# ------------------------------------------------------------------------------
# curl
# ------------------------------------------------------------------------------

set(BUILD_TESTING OFF)
set(HTTP_ONLY ON)
set(CMAKE_USE_LIBSSH2 OFF)

include_directories(${PROJECT_SOURCE_DIR}/lib/curl/include)

if (MSVC)
	set(CURL_STATICLIB true)
	set(CURL_STATIC_CRT ON)
	set(BUILD_CURL_EXE false)
	set(BUILD_CURL_EXE false)
	add_subdirectory(${PROJECT_SOURCE_DIR}/lib/curl)
endif()

if (APPLE)
  target_link_libraries(${PROJECT_NAME} PUBLIC CURL)
else()
	target_link_libraries(${PROJECT_NAME} PUBLIC libcurl)
endif()

# ------------------------------------------------------------------------------
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,8 @@ set(CMAKE_CXX_STANDARD 11)
 
 set(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR})
 
+set(BUILD_SHARED_LIBS OFF)
+
```
Comment (sleepdefic1t) [inline CMakeLists.txt]:
```cmake
if (APPLE)
  set(OPENSSL_ROOT_DIR "/usr/local/opt/openssl/lib")
  set(OPENSSL_INCLUDE_DIR "/usr/local/opt/openssl/include")
endif()
```

**Comment by sleepdefic1t** [comment]
@ciband 

I got this to build and passed tests on my system (macOS) with a few tweaks.

I changed a couple parts and ran `brew install openssl`.
(changes in PR comments; the brew bit just add to `config.yml`).

Thank you for working on this.
Much appreciated, sir!

**Comment by ciband** [comment]
Do thoses changes force the apple builds to use an installed version of openssl instead of our internally build version?

Is that desirable vs. all platforms running the same thing?

**Comment by sleepdefic1t** [comment]
@ciband 
It totally does use an installed version.

Yeah, it'd be preferred to have all builds do the same thing.

If you have anything you'd like me to try, just let me know 👍

**Comment by sleepdefic1t** [comment]
@faustbrian 
Can you retrigger Appveyor when you get a moment, please and thank you.

**Comment by faustbrian** [comment]
@ciband conflicts

**Comment by faustbrian** [comment]
https://ci.appveyor.com/project/faustbrian/cpp-client/builds/27587615/job/etm2hrawtocnsp6q

**Comment by ciband** [comment]
@faustbrian  same issue here as in cpp-crypto.  I fixed the conflict and merged develop into the PR.  When I go to push I get an error.  Will investigate later.

**Comment by sleepdefic1t** [comment]
@ciband 
#132 May help kick start CI Actions after it's merged and pulled in.

If not, you may have to redo the PR from a fresh fork.
Hopefully we can avoid that though 👍

**Comment by ciband** [comment]
Looks like the same Appveyor issues.  With the changes the code builds, which was the main problem.  CI will be fixed in a github windows ci commit.

**Pull request merged**